src/Orchard.Web/Modules/Orchard.Tokens/Providers/ListTokens.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web;
+using Orchard.ContentManagement;
+using Orchard.Localization;
+using System.Linq;
+	 
+namespace Orchard.Tokens.Providers {
+	 
+	public class ListTokens : ITokenProvider {
+	    private readonly Func<ITokenizer> _tokenizer;
+	 
+	    public ListTokens(Func<ITokenizer> tokenizer) {
+	        _tokenizer = tokenizer;
+	        T = NullLocalizer.Instance;
+	    }
+	 
+	    public Localizer T { get; set; }
+	 
+	    public void Describe(DescribeContext context) {
+	        context.For("List", T("Lists"), T("Handles lists of Content Items"))
+                .Token("Join:*", T("Join:<Tokenized text>[,<separator>]"), T("Join each element in the list by applying the tokenized text and concatenating the output with the optional separator."))
+                .Token("SumInt:*", T("SumInt:<Tokenized text>"), T("Sum each element in the list by applying the tokenized text."))
+                .Token("SumFloat:*", T("SumFloat:<Tokenized text>"), T("Sum each element in the list by applying the tokenized text."))
+                .Token("SumDecimal:*", T("SumDecimal:<Tokenized text>"), T("Sum each element in the list by applying the tokenized text."))
```

Comment:
I would rather see a single Sum which handles decimal, then use a different token if it needs to be rounded or formatted.